Library/Formula/mlkit.rb

```diff
@@ -0,0 +1,23 @@
+class Mlkit < Formula
+  desc "Compiler for the Standard ML programming language"
+  homepage "https://melsman.github.io/mlkit"
+  url "https://github.com/melsman/mlkit/archive/mlkit-4.3.9.tar.gz"
+  sha256 "3c6adbeb9a85f7b3586d0961fd3b170ff31e09fa0ff12889b76b9ceb459059c4"
+
+  depends_on "autoconf" => :build
+  depends_on "mlton" => :build
+  depends_on "gmp"
+
+  def install
+    system "./autobuild; true"
+    system "./configure", "--prefix=#{prefix}"
+    ENV.permit_arch_flags
```

Comment:
The permit_arch_flags is needed because the mlkit generates 32-bit machine
code whereas the mlton compiler generates 64-bit code. Because of this
aspect, the ENV.m64 and ENV.m32 flags are not sufficient for the formula as
clang is used by both tools in a single makefile target.

> In Library/Formula/mlkit.rb
> <https://github.com/Homebrew/homebrew/pull/49933#discussion_r55938866>:
>
> > @@ -0,0 +1,23 @@
> > +class Mlkit < Formula
> > +  desc "Compiler for the Standard ML programming language"
> > +  homepage "https://melsman.github.io/mlkit"
> > +  url "https://github.com/melsman/mlkit/archive/mlkit-4.3.9.tar.gz"
> > +  sha256 "3c6adbeb9a85f7b3586d0961fd3b170ff31e09fa0ff12889b76b9ceb459059c4"
> > +
> > +  depends_on "autoconf" => :build
> > +  depends_on "mlton" => :build
> > +  depends_on "gmp"
> > +
> > +  def install
> > +    system "./autobuild; true"
> > +    system "./configure", "--prefix=#{prefix}"
> > +    ENV.permit_arch_flags
>
> Can you explain why this is needed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/49933/files#r55938866>.
>